docs/wip/theme-shadowing.md

```diff
@@ -1,7 +1,82 @@
+
+In each of the following "how to shadow" examples, we will present a goal and show how to achieve it:
+
+1. We want to completely shadow the [AboveTheFold](https://github.com/vtex/faststore/blob/master/packages/gatsby-theme-store/src/components/HomePage/AboveTheFold.tsx) component of the HomePage in our example-store.
+
+    Firstly, we create a new file named `AboveTheFold.tsx`, just like the original file. This will result in the following directory tree:
+    ```
+    storecomponents.store
+    └── src
+        └── @vtex
+            └──gatsby-theme-store
+                └── components
+                    └── HomePage
+                        └──AboveTheFold.tsx
+    ```
+    Then, we implement it however we want. In this example, we go from [this](https://github.com/vtex/faststore/blob/master/packages/gatsby-theme-store/src/components/HomePage/AboveTheFold.tsx), which is just a text explaining what should be in this _"fold"_:
+    ```
+    const Fold: FC<Props> = () => (
+      <Center height="800px">
+        <Text sx={{ width: '50%' }}>
+          This is the Above the fold part of your home page. All sync items should
+          be rendered in here. Thus, make sure all data rendered in this part is
+          fetched during Server Side Rendering and revalidated on the client if
+          necessary
+        </Text>
+      </Center>
+    )
+    ```
+    to [this](https://github.com/vtex-sites/storecomponents.store/blob/master/src/%40vtex/gatsby-theme-store/components/HomePage/AboveTheFold.tsx) implementation:
+    ```
+    const Fold: FC<Props> = () => (
+      <>
+        <Carousel allItems={CAROUSEL_ITEMS} height="540px" width="360px" />
+        <RichTextRow />
+      </>
+    )
+    ```
+
+2. We want to customize, but also reuse the code from the original file.
+
+    In order to do that, we would create a file like the previous example, but now we would import the theme's original file. That way, we won't have to copy & paste all the code that we will reuse, we will only be concerned with the new code that we are going to implement.
+
+    This is shown in [this](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/#importing-the-shadowed-component) example from the Gatsby documentation, in which a component is imported and wrapped with another component. 
+    In addition, there is an [example](https://github.com/vtex-sites/storecomponents.store/blob/master/src/%40vtex/gatsby-theme-store/components/ProductSummary/index.tsx) in our example-store, where we import the `Props` interface from the [shadowed file](https://github.com/vtex/faststore/blob/master/packages/gatsby-theme-store/src/components/ProductSummary/index.tsx) to use it in our brand new ProductSummary component.
 
 ## Reference Implementation
-Remember to reference public repos
+Since Gastby is an open source framework, that are many examples out there, but you could always refer to our [example-store](https://github.com/vtex-sites/storecomponents.store).
```

Comment:
```suggestion
Since Gatsby is an open-source framework, there are many examples out there, but you could always refer to our [Store Component starter (`storecomponents.store`)](https://github.com/vtex-sites/storecomponents.store).
```